### PR TITLE
Fix the OSBMS scaling KUTTL test

### PR DESCRIPTION
**Pull request by abays**
Fixes for the `OpenStackBaremetalSet` scaling KUTTL test.  Also adjusts the default RHEL image in our config samples and makes minor edits to other KUTTL tests that consider this image.

### CODE REVIEW

File: tests/kuttl/tests/openstackbaremetalset_scale/09-assert.yaml [hunk 1]
```diff
@@ -32,21 +32,3 @@ status:
...
-commands:
-  - script: |
-      set -x
-      sleep 5
-      osbmsstate=$(oc get osbms -n openstack compute -o json | jq -r '.status.provisioningStatus.reason')
-      if grep -q "Please check BaremetalHost resources and re-add" <<< "$osbmsstate"; then
-        exit 0
-      else
-        exit 1
-      fi
```
Comment (abays) [inline tests/kuttl/tests/openstackbaremetalset_scale/09-assert.yaml:52]: This was unnecessary.  Just checking the OSBMS `status.provisioningStatus.readyCount` and `status.provisioningStatus.state` is enough to detect the problem we are expecting.

File: tests/kuttl/tests/openstackbaremetalset_scale/11-assert.yaml [hunk 2]
```diff
@@ -33,7 +33,4 @@ status:
       hostname: compute-1
   provisioningStatus:
     readyCount: 1
-    reason: Existing BaremetalHost "openshift-worker-0" not found for OpenStackBaremetalSet
-      compute.  Please check BaremetalHost resources and re-add "openshift-worker-0"
-      to continue
```
Comment (abays) [inline tests/kuttl/tests/openstackbaremetalset_scale/11-assert.yaml:38]: This was unnecessary.  Just checking the OSBMS `status.provisioningStatus.readyCount` and `status.provisioningStatus.state` is enough to detect the problem we are expecting.

File: tests/kuttl/common/scripts/delete_bmh.sh [hunk 3]
```diff
@@ -4,11 +4,12 @@
 # Dumps a BMH and its credentials secret to designated locations and then deletes the BMH
 #
 
-bmhname=$(oc get -n openstack osbms compute -o json | jq -r '.status.baremetalHosts["compute-0"].hostRef')
-bmhsecretname=$(oc get bmh -n openshift-machine-api ${bmhname} -o json | jq -r '.spec.bmc.credentialsName')
-oc get -n openshift-machine-api bmh ${bmhname} -o json |\
+bmhname=$(oc get bmh -n openshift-machine-api -l osp-director.openstack.org/name=compute -l osp-director.openstack.org/namespace=openstack -o name | head -1)
+bmhsecretname=$(oc get -n openshift-machine-api ${bmhname} -o json | jq -r '.spec.bmc.credentialsName')
```
Comment (abays) [inline tests/kuttl/common/scripts/delete_bmh.sh:8]: We don't need to pick a specific host/BMH.  We just need to delete one of them, and so we now just choose the first one in the list.

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**